packages/@aws-cdk/aws-ecs/lib/fargate/fargate-service.ts

```diff
@@ -72,10 +67,47 @@ export class FargateService extends BaseService {
 
     this.configureAwsVpcNetworking(props.cluster.vpc, props.assignPublicIp, props.vpcSubnets, props.securityGroup);
 
+    if (props.serviceDiscoveryOptions) {
+      this.enableServiceDiscovery(props.serviceDiscoveryOptions);
+    }
+
     if (!props.taskDefinition.defaultContainer) {
       throw new Error('A TaskDefinition must have at least one essential container');
     }
   }
+
+  private enableServiceDiscovery(options: ServiceDiscoveryOptions): cloudmap.Service {
```

Comment:
Can this not go on `BaseService`? What's the difference between the 2 implementations?